### PR TITLE
replace sort.Slice with slices.Sort, slices.SortFunc

### PR DESCRIPTION
**Pull request by justlorain**
Replace `sort.Slice` with `slices.Sort` and `slices.SortFunc`. 
As mentioned in the comments of `sort.Slice`, the newer method is more ergonomic and run faster.

### CODE REVIEW

**Review by pav-kv — APPROVED** [review]
LGTM, though you might want to get a review from a maintainer. CC @ahrtr

**Comment by ahrtr** [comment]
@justlorain could you please signoff the commit? 

```
git rebase HEAD~1 --signoff
git push --force-with-lease origin optimize/replace_sort
```

**Pull request merged**
